### PR TITLE
feat: 🎸 Always return `TickerReservation` from `assets.getTickerReservation`

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

Regardless of the state, `assets.getReservation` now returns a `TickerReservation` instance.

### Breaking Changes

🧨 `assets.isTickerAvailable` has been removed. 
To check ticker availability, directly get the `TickerReservation` instance and use `details.status` to check the availability

### JIRA Link

DA-142

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/Assets.ts [hunk 1]
```diff
@@ -149,32 +114,9 @@ export class Assets {
    */
   public async getTickerReservation(args: { ticker: string }): Promise<TickerReservation> {
```
Comment (polymath-eric) [inline src/Assets.ts:115]: Should we keep this async? looks like the await got removed.

**Review by polymath-eric — APPROVED** [review]
LGTM

File: src/utils/internal.ts [hunk 2]
```diff
@@ -1033,3 +1034,31 @@ export function assertExpectedChainVersion(nodeUrl: string): Promise<void> {
     };
   });
 }
+
+/**
+ * @hidden
+ *
+ * Validates a ticker value
+ */
+export function assertTickerValid(ticker: string) {
```
Comment (monitz87) [inline src/utils/internal.ts:1043]: Missing return type

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**